database/postgres/postgres.go

```diff
@@ -266,13 +277,23 @@ func runesLastIndex(input []rune, target rune) int {
 	return -1
 }
 
+func (p *Postgres) quoteIdentifier(name string) string {
+	if p.config.MigrationsTableHasSchema {
+		firstDotPosition := strings.Index(name, ".")
+		if firstDotPosition != -1 {
+			return pq.QuoteIdentifier(name[0:firstDotPosition]) + "." + pq.QuoteIdentifier(name[firstDotPosition+1:])
```

Comment:
We error if more than one `.` is found or document the behavior. e.g. the first dot is treated as the schema and table name separator.

Another approach I just thought of is to have an option to treat the identifier as already quoted. e.g. you'd pass in `"myschema"."mytable"` and we'd just check that the string starts and ends with `"`. Thoughts?

Reply:
>  or document the behavior. e.g. the first dot is treated as the schema and table name separator.

@dhui I chose to document it.

Reply:
> Another approach I just thought of is to have an option to treat the identifier as already quoted. e.g. you'd pass in "myschema"."mytable" and we'd just check that the string starts and ends with ". Thoughts?

Do you think that this is a better approach ?

Reply:
I think having an option to treat the identifier as already quoted and requiring that the identifier starts and ends with a `"` is a simpler approach since we're not parsing the identifer so there are fewer opportunities for unexpected behaviors
What do you think?

If we go with the pre-quoted identifier approach, what do you think of the config name `x-migrations-table-quoted`?

Reply:
> I think having an option to treat the identifier as already quoted and requiring that the identifier starts and ends with a " is a simpler approach since we're not parsing the identifer so there are fewer opportunities for unexpected behaviors
What do you think?

@dhui This is a new Pull Request with this implementation: https://github.com/golang-migrate/migrate/pull/533